gcc/rust/util/rust-attributes.cc

```diff
@@ -67,69 +67,60 @@ using Attrs = Values::Attributes;
 
 // https://doc.rust-lang.org/stable/nightly-rustc/src/rustc_feature/builtin_attrs.rs.html#248
 static const BuiltinAttrDefinition __definitions[]
-  = {{Attrs::INLINE, CODE_GENERATION},
-     {Attrs::COLD, CODE_GENERATION},
-     {Attrs::CFG, EXPANSION},
-     {Attrs::CFG_ATTR, EXPANSION},
-     {Attrs::DERIVE_ATTR, EXPANSION},
-     {Attrs::DEPRECATED, STATIC_ANALYSIS},
-     {Attrs::ALLOW, STATIC_ANALYSIS},
-     {Attrs::ALLOW_INTERNAL_UNSTABLE, STATIC_ANALYSIS},
-     {Attrs::DOC, HIR_LOWERING},
-     {Attrs::MUST_USE, STATIC_ANALYSIS},
-     {Attrs::LANG, HIR_LOWERING},
-     {Attrs::LINK_NAME, CODE_GENERATION},
-     {Attrs::LINK_SECTION, CODE_GENERATION},
-     {Attrs::NO_MANGLE, CODE_GENERATION},
-     {Attrs::REPR, CODE_GENERATION},
-     {Attrs::RUSTC_BUILTIN_MACRO, EXPANSION},
-     {Attrs::RUSTC_MACRO_TRANSPARENCY, EXPANSION},
-     {Attrs::PATH, EXPANSION},
-     {Attrs::MACRO_USE, NAME_RESOLUTION},
-     {Attrs::MACRO_EXPORT, NAME_RESOLUTION},
-     {Attrs::PROC_MACRO, EXPANSION},
-     {Attrs::PROC_MACRO_DERIVE, EXPANSION},
-     {Attrs::PROC_MACRO_ATTRIBUTE, EXPANSION},
-
-     {Attrs::DERIVE, EXPANSION},
+  = {{Attrs::INLINE, CODE_GENERATION, false},
```

Comment:
I think it would be better to add a second map here rather than add a boolean attribute to every BuiltinAttrDefinition